frappe/public/js/frappe/form/footer/timeline.js

```diff
@@ -128,7 +128,33 @@ frappe.ui.form.Timeline = Class.extend({
 				var name = $(this).parents(".timeline-item:first").attr("data-name");
 				me.delete_comment(name);
 				return false;
-			});
+			})
+            .on('click', '.edit', function() {
+                var is_editing = 'is-editing';
```

Comment:
I used the check-icon solution for now.
Removing the textarea border seemed somewhat out of place and leaving a missing hint that you can click in it for editing now. Could make it thinner though.